runtime/lua/vim/lsp/_watchfiles.lua

```diff
@@ -6,13 +6,11 @@ local lpeg = vim.lpeg
 
 local M = {}
 
----@alias lpeg userdata
-
 --- Parses the raw pattern into an |lpeg| pattern. LPeg patterns natively support the "this" or "that"
 --- alternative constructions described in the LSP spec that cannot be expressed in a standard Lua pattern.
 ---
 ---@param pattern string The raw glob pattern
----@return lpeg An |lpeg| representation of the pattern, or nil if the pattern is invalid.
+---@return vim.lpeg.Pattern? pattern An |lpeg| representation of the pattern, or nil if the pattern is invalid.
```

Comment:
```suggestion
---@return vim.lpeg.Pattern? pattern An |lpeg| representation of the pattern, or nil if the pattern is invalid or not found.
```

Reply:
Why not found? Isn't this referring to implementation detail of the internally used capture group?

Reply:
I just feel like this description makes it sound like `nil` will be returned when the input is invalid, but it can also me returned if the match fails.

...Which I guess means that the input isn't a valid pattern. Nvm, ignore my previous comment.